### DISCUSSION TITLE
FYI: Question asked in Quarkus repo re FalkorDB support

### DISCUSSION THREAD

**Discussion by flowt-au** [FalkorDB/FalkorDB#638] [General]
Hi. Just a FYI. I just posted a question in the [Quarkus repo](https://github.com/quarkusio/quarkus/issues/40092) about their intensions regarding FalkorDB.

Any thoughts from your side?

Thanks again,
Murray

**Comment by gkorland** [comment]
Redis COPY is not supported by FalkorDB only `GRAPH.COPY` see: https://docs.falkordb.com/commands/graph.copy.html

**Reply by flowt-au** [reply]
Thanks Guy. So given my app uses the[ Quarkus Reactive Redis client](https://quarkus.io/guides/redis-reference#use-the-vert-x-redis-client), which wraps [Vertx Redis Client](https://vertx.io/docs/vertx-redis-client/java/), I am thinking I will need to use the "low level" Vertx connection directly to get the new Falkor functionality like GRAPH.COPY and the vector db functionality. 

Anyway, I will give it a go and see what happens.
Thanks,
Murray

**Reply by gkorland** [reply]
Can't you use JFalkorDB?
https://github.com/FalkorDB/JFalkorDB/

**Reply by flowt-au** [reply]
Possibly, but I have written an entire database wrapper around RedisGraph (and now using FalkorDB instead) that leverages the Quarkus Reactive approach (ie using SmallRye Uni / Multi) and the Vertx Reactive Redis client, as mentioned. It works really well, except that now I want to start using the FalkorDB functionality that Redis (Vertx) knows nothing about. To use `JFalkorDB` I would need to work out how to wrap it with a Uni so it doesn't block the thread on the Uni pipeline. My Quarkus app is an API endpoint which builds a Uni pipeline for which FalkorDB is the "database" layer. I am sure that is possible but will be a big learning curve for me. And, if that is what it takes, so be it. I really like both Quarkus and FalkorDB and want to continue to use them together.

**Reply by flowt-au** [reply]
And maybe that isnt so hard

```java
Uni<T> myAsyncMethod() {
    return Uni.createFrom().item(() -> {
        // Blocking third-party call
        T result = blockingThirdPartyMethod();
        return result;
    }).runSubscriptionOn(Infrastructure.getDefaultExecutor());
}
```

Anyway....

**Reply by gkorland** [reply]
I'm not that familiar with Quarkus, so I can't comment about your last comment. Does it work for you?

**Reply by flowt-au** [reply]
I am just about to start my day and will do some tests. If I can wrap JFalkorDB in a Uni then I can make a reactive FalkorDB client and use that instead of the Quarkus reactive Redis client. The upside of that is (apart from working with FalkorDB!) it will be more light-weight because it wont have the code needed to support the other Redis modules.

I will get back to you. Thanks!
